transcom/mymove
# [GHC] :bug: Fix Null GBLOC in Support API

## Summary

The support API for creating a move task order did not include logic to set the GBLOC. Since the Prime regularly uses this endpoint to generate moves for their testing, this was causing issues downstream due to a null GBLOC. This fix fetches the appropriate GBLOC value based on the origin duty location address.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Run the server locally or point to an ephemeral build API using your tooling of choice (cURL, Postman, etc.)
2. Use the Support API to [Create A Move Task Order](https://transcom.github.io/mymove-docs/api/support#tag/moveTaskOrder/operation/createMoveTaskOrder) - NOTE you will want use your local DB or utilize the network tab to copy the following UUID values from an existing move:
    1. `OriginDutyLocationID`
    2. `DestinationDutyLocationID`
    3. `OrdersUploadID`
3. Log in to the office app as a service counselor in the same GBLOC associated with the Origin Duty Location and confirm that the move shows up in their queue OR Query the move in the database to verify that the `gbloc` value is correctly set
    1. Example query: 
    ```
        SELECT o.gbloc, * 
        FROM moves m 
        INNER JOIN orders o ON m.orders_id = o.id
        WHERE m.locator = '8K8Y43'
    ```
